src/stronghold.hs

```diff
@@ -228,7 +236,14 @@ site runStoreOp =
     json <- runStoreOp $ do
       hier <- Trees.lastHierarchy hist
       snd <$> Trees.materializedView path hier
+
+    let etag = (Base16.encode . hash . toStrict . Aeson.encode) json
+
+    Snap.modifyResponse $ Snap.setHeader etagHeader etag
     Snap.writeLBS $ Aeson.encode json
+   where
+     -- BS.toStrict is available in more moderns versions of GHC
+     toStrict = fromMaybe BS.empty . listToMaybe . LBS.toChunks
```

Comment:
`listToMaybe` seems to be a safe `head`. It'll take a list, if there are any elements it'll return `Just` the first one, otherwise `Nothing`. In your case the list are the chunks from the lazy bytestrings, of which there might be more than one! If you do want to convert the lazy bs to a strict one, `BS.concat . LBS.toChunks` is the way to go